packages/runtime-common/indexer.ts

```diff
@@ -668,14 +714,30 @@ export class Indexer {
     if (!api) {
       throw new Error(`could not load card API`);
     }
-    let fields = api.getFields(cardOrField);
-    let field = fields[currentSegment];
-    if (!field) {
-      throw new Error(
-        `Your filter refers to nonexistent field "${currentSegment}" on type ${JSON.stringify(
-          identifyCard(cardOrField),
-        )}`,
-      );
+    let field: Field;
+    if (currentSegment === '_cardType') {
+      // this is a little awkward--we have the need to treat the card type as a
+      // type of string field that we can query against from the index (e.g. the
+      // cards grid sorts by the card's display name). current-run is injecting
+      // this into the searchDoc during index time.
+      field = {
+        card: (
+          await loader.import<{ default: typeof CardAPI.FieldDef }>(
+            'https://cardstack.com/base/string',
+          )
+        ).default,
+        fieldType: 'contains',
+      } as unknown as Field; // just pretend this is an actual field
```

Comment:
FYI this was an interesting situation i can into. hopefully there is not more data like this. if so we should probably make a `_meta` field in the search doc to hold this kind of stuff